hugolib/site.go

```diff
@@ -105,7 +106,8 @@ type SiteInfo struct {
 	Authors               AuthorList
 	Social                SiteSocial
 	Sections              Taxonomy
-	Pages                 *Pages
+	Pages                 *Pages // Includes only pages in this language
+	TranslatedPages       *Pages // Includes other translated pages, excluding those in this language.
```

Comment:
since we have `.Pages` already, and `setupTranslations` tweaks it to remove pages from the other language, we'd need to make `.AllPages` the main reference throughout the code in `site.go`, and in `setupTranslations`, filter out the non-current-language pages and put them in `.Pages`.

A little more involved..